openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -775,13 +788,350 @@ function scale(IndicatorObj) {
+            // primary indicator weight, the map must be refreshed.
+            // Example 2: If the user is viewing a theme thematic map (e.g. Economy), and then changes
+            // the SVI weight, the map should not be redrawn.
+        // 2) the thematic map menu has been changes
+        // 3) a new project is loaded into the map (case: weightChange = 0)
+    var selectedIndicatorLabel = $('#webGlThematicSelection option:selected').attr('class');
+
+    if (weightChange < parseInt(selectedIndicatorLabel || selectedIndicatorLabel !== '4')) {
+        if (weightChange !== 0) {
+            $('#absoluteSpinner').hide();
+            return;
+        }
+    }
+
+    // Find the values to create categorized color ramp
+    // First find the min and max vales
+    var minMaxArray = [];
+    for (var i = 0; i < mappingLayerAttributes.features.length; i++) {
+        var match = false;
+        for (var k in mappingLayerAttributes.features[i].properties) {
+            if (k == selectedIndicator) {
+                match = true;
+                minMaxArray.push(mappingLayerAttributes.features[i].properties[k]);
+                break
+            }
+            else {
+                match = false;
+            }
+        }
+        if (match === false) {
+            return;
+        }
+    }
+
+    // Focus map on layer
+    if (projectChange) {
+        map.fitBounds(mapboxBoundingBox);
+    }
+
+    var min = Math.min.apply(null, minMaxArray).toFixed(2);
+    var max = Math.max.apply(null, minMaxArray).toFixed(2);
+    min = (parseFloat(min) - 0.1);
+    // Round up the max
+    max = Math.ceil(max * 10) / 10;
+
+    // TODO allow the user to change the colors
+    var colorsPalRed = ['#fee5d9', '#fcbba1', '#fc9272', '#fb6a4a', '#de2d26', '#a50f15'];
+    var colorsPalBlue = ['#eff3ff', '#c6dbef', '#9ecae1', '#6baed6', '#3182bd', '#08519c'];
+    var colorsPal = colorsPalBlue;
+    var breaks = [];
+
+    function getColor() {
+        var interval = (max - min) / 6;
+        var tempStep = min;
+        for (var i = 0; i < 5; i++) {
+            tempStep += interval
+            breaks.push(tempStep);
+        }
+        breaks.unshift(min);
+        breaks.push(max);
+    }
+
+    getColor();
+
+    // Case 2, and 4, try to remove any existing layers
+    // TEMP also case 3
+    //if (weightChange === false) {
+        try {
+            for (var i = 0; i < 6; i++) {
+                map.removeLayer(i);
+            };
+        } catch (e) {
+            // continue
+        }
+    //}
+
+    // Case 3 update the source
+    if (weightChange) {
+        //console.log('Update the source:');
+        //map.update(true);
+
+        // FIXME would really prefer to update the map source here
+        map.removeSource('projectSource');
+    }
+
+    // Case 4 destory the source
+    if (projectChange) {
+        try {
+            map.removeSource('projectSource');
+        } catch (e) {
+            // continue
```

Comment:
Ok, perhaps you could add a short comment to the code, to make it more explicit.